### PR TITLE
Add some I2C commands

### PR DESCRIPTION
**Pull request by sjg20**
Add some I2C commands to read and write bytes

Again I am not sure how to test these. I don't see a fake I2C driver for posix.

### CODE REVIEW

**Comment by sglass68** [comment]
Hi, just checking that you saw this one

**Comment by nashif** [comment]
@sglass68  needs a rebase

File: drivers/i2c/i2c_shell.c [hunk 1]
```diff
@@ -87,6 +91,130 @@ static int cmd_i2c_recover(const struct shell *shell,
...
+		shell_error(shell, "I2C: Device driver %s not found.", argv[1]);
+		return -ENODEV;
+	}
+
+	dev_addr = strtol(argv[2], NULL, 16);
+	reg_addr = strtol(argv[3], NULL, 16);
+	num_bytes = argc - 4;
+	if (num_bytes < 0)
+		return 0;
+	if (num_bytes > MAX_I2C_BYTES)
```
Comment (nashif) [inline drivers/i2c/i2c_shell.c:115]: need curly braces around if statement

File: drivers/i2c/i2c_shell.c [hunk 2]
```diff
@@ -87,6 +91,130 @@ static int cmd_i2c_recover(const struct shell *shell,
...
+		num_bytes = MAX_I2C_BYTES;
+	for (i = 0; i < num_bytes; i++)
```
Comment (nashif) [inline drivers/i2c/i2c_shell.c:117]: need curly braces around for statement

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by nashif** [comment]
> @sglass68 needs a rebase

just did a rebase.

File: drivers/i2c/i2c_shell.c (hunk 1 above)
Comment (sglass68) [reply drivers/i2c/i2c_shell.c]: Is there a way to check for this? Checkpatch doesn't warn me and I'm not used to it as U-Boot, kernel, etc. are different.

**Review by nashif — APPROVED** [review]

File: drivers/i2c/i2c_shell.c (hunk 1 above)
Comment (carlescufi) [reply drivers/i2c/i2c_shell.c]: checkpatch should definitely warn you about this. Maybe there's a bug in it.

**Review by carlescufi — APPROVED** [review]
LGTM, nice addition, thanks!

**Pull request merged**